google-cloud-spanner/src/main/java/com/google/cloud/spanner/GrpcStruct.java

```diff
@@ -193,6 +216,11 @@ public String toString() {
 
   boolean consumeRow(Iterator<com.google.protobuf.Value> iterator) {
     rowData.clear();
+    if (decodeMode == DecodeMode.LAZY_PER_ROW) {
+      rowDecoded = false;
+    } else if (decodeMode == DecodeMode.LAZY_PER_COL) {
+      colDecoded.clear();
+    }
```

Comment:
nitpick: I guess we can also reset these values unconditionally

Reply:
No (at least not in the current setup), as `colDecoded` is `null` in all cases, except when `decodeMode == DecodeMode.LAZY_PER_COL`. We only initialize that bitset if necessary (see line 194).